File: frontend/src/services/fir.ts
import { api } from './api';

export interface FIRResponse {
  id: number;
  fir_number: string;
  incident_date: string;
  registration_date: string;
  status: string;
  summary: string | null;
  station_name: string;
  category_name: string;
}

export interface FIRPaginatedResponse {
  total_count: number;
  items: FIRResponse[];
}

export const firService = {
  searchFirs: async (
    skip: number = 0,
    limit: number = 50,
    searchQuery?: string,
    statusFilter?: string
  ): Promise<FIRPaginatedResponse> => {
    // Build the query parameters dynamically
    const params = new URLSearchParams({
      skip: skip.toString(),
      limit: limit.toString(),
    });
    
    if (searchQuery) params.append('search', searchQuery);
    if (statusFilter && statusFilter !== 'All') params.append('status', statusFilter);

    const response = await api.get(`/firs/?${params.toString()}`);
    return response.data.data; // Unpack the nested FastAPI response
  }
};

File: backend/app/schemas/fir.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import date

class FIRResponse(BaseModel):
    id: int 
    fir_number: str
    incident_date: date
    registration_date: date
    status: str
    summary: Optional[str] = None
    station_name: str
    category_name: str

class FIRPaginatedResponse(BaseModel):
    total_count: int
    items: List[FIRResponse]